### PR TITLE
Add button for user to navigate themselves to nearby place

### LINKED ISSUE #267: Integrate other navigation apps
It would be nice if when a user clicked on a place details it populated their navigation app of choice with directions there. Apps could include Lazarillo, Google Maps, and Apple Maps

### PR DESCRIPTION
**Pull request by carpenterd777**
Closes #267, depends on #272

Testing on Apple Maps was not possible for me, so I am relying on @aej11a, @jlora23, and @elenirotsides for feedback on how it looks on your devices

Adds the `MapAnchor` component, which is used to create pressable elements that when pressed open the user's navigation app of choice with navigation opened directing them how to walk to that place.

### CODE REVIEW

File: src/components/MapAnchor.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	destination_place_id,
+	formatted_address,
+	children,
+}: React.PropsWithChildren<MapAnchorProps>) => {
+	const MAPS_HREF = buildMapsHref(
+		TravelMode.Walking,
+		destination_place_id,
+		destination,
+		formatted_address
+	);
```
Comment (aej11a) [inline src/components/MapAnchor.tsx:94]: not a hard requirement but generally we've been preferring object args

File: src/components/MapAnchor.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+const buildGoogleMapsHref = (
+	travelmode: TravelMode,
+	startNavigation: boolean,
+	destination?: string,
+	destination_place_id?: string
+) => {
+	// https://developers.google.com/maps/documentation/urls/get-started#directions-action
+	let href = "https://www.google.com/maps/dir/?api=1";
+	href += `&travelmode=${travelmode}`;
```
Comment (aej11a) [inline src/components/MapAnchor.tsx:37]: sort of a nitpick but prob should either add or use the template literal here, not both
`let href = https://www.google.com/maps/dir/?api=1&travelmode=${travelmode}`

Comment (aej11a) [reply src/components/MapAnchor.tsx:37]: or 
	let href = "https://www.google.com/maps/dir/?api=1&travelmode=";
	href += travelmode;

or
	let href = "https://www.google.com/maps/dir/?api=1";
	href += `&travelmode=`;
	href += travelmode;
your call but that was my gut instinct, just seemed a little weird to add something on the next line with a template literal in it

**Review by aej11a — CHANGES_REQUESTED** [review]

File: src/components/MapAnchor.tsx (hunk 1 above)
Comment (carpenterd777) [reply src/components/MapAnchor.tsx:94]: any particular reason why or just code style?

Comment (aej11a) [reply src/components/MapAnchor.tsx:94]: just code style - i don't really care that much about this but i think i remember we both agreed that Eleni should switch some params to an objecte on one of her earlier PRs, so i wanted to have consistent governance

**Review by aej11a — APPROVED** [review]

**Pull request merged**